httprange_test.go

```diff
@@ -52,16 +55,36 @@ func TestParseRequestRange(t *testing.T) {
 	}
 
 	// Test invalid range strings.
-	invalidRangeStrings := []string{"", "2-5", "bytes= 2-5", "bytes=2 - 5", "bytes=0-0,-1", "bytes=5-2", "bytes=2-5 ", "bytes=2--5"}
+	invalidRangeStrings := []string{
+		"bytes=8",
+		"bytes=5-2",
+		"bytes=+2-5",
+		"bytes=2-+5",
+		"bytes=2--5",
+		"bytes=-",
+		"",
+		"2-5",
+		"bytes = 2-5",
+		"bytes=2 - 5",
+		"bytes=0-0,-1",
+		"bytes=2-5 ",
+	}
 	for _, rangeString := range invalidRangeStrings {
 		if _, err := parseRequestRange(rangeString, 10); err == nil {
 			t.Fatalf("expected: an error, got: <nil>")
 		}
 	}
 
 	// Test error range strings.
-	errorRangeString := "bytes=-0"
-	if _, err := parseRequestRange(errorRangeString, 10); err != errInvalidRange {
-		t.Fatalf("expected: %s, got: %s", errInvalidRange, err)
+	errorRangeString := []string{
```

Comment:
The error cases and the succees should be formed into a single array. They should be together, the success and error cases shouldn't be separated.